### ISSUE TITLE
Regression in PolygonSpatialObject::IsInsideInObjectSpace

### ISSUE THREAD

**Issue by maekclena** [InsightSoftwareConsortium/ITK#1082]
### Description

Polygon spatial objects have extra horizontal lines on some vertices. It is especially apparent when converted to binary image.

### Steps to Reproduce

```c++
#include <itkImage.h>
#include <itkPolygonSpatialObject.h>

using ImageType = itk::Image<unsigned char, 2>;
using PolygonType = itk::PolygonSpatialObject<2>;

int main()
{
  PolygonType::Pointer polygon = PolygonType::New();

  // bottom left triangle
  float points[][2] = {
    {0.0, 0.0},
    {0.0, 10.0},
    {10.0, 10.0},
  };

  for (auto point : points)
  {
    PolygonType::SpatialObjectPointType so_point;
    so_point.SetPositionInObjectSpace(point);
    polygon->AddPoint(so_point);
  }

  polygon->Update();

  // top right corner
  ImageType::PointType test_point;
  test_point[0] = 10.0;
  test_point[1] = 0.0;

  // outputs "1"
  std::cout << polygon->IsInsideInObjectSpace(test_point, 0, "") << std::endl;

  return 0;
}
```

### Expected behavior

`test_point` shouldn't be inside the polygon.

### Actual behavior

`test_point` is inside the polygon.

### Reproducibility

100%

### Versions

The regression was introduced in b83454ef2b4d2385adddab67d6378fe8c47f8800.

### Environment

MacOS Mojave

**Comment by neslimsah** [comment]
Hello,

I faced the same issue. I resolved it by the following update at itkPolygonialSpatialObject.hxx

Before Update:
![beforeUpdate](https://user-images.githubusercontent.com/1110802/62486936-73599300-b78e-11e9-88f8-90191288fb10.png)

After Update:
![afterUpdate](https://user-images.githubusercontent.com/1110802/62486935-73599300-b78e-11e9-8d89-6e33757fec22.png)

I colored the contour points in the ascending order in the list. 

For my local problem, the problem occurs at:
<img width="327" alt="image" src="https://user-images.githubusercontent.com/1110802/62487241-507bae80-b78f-11e9-896b-3051e5f87a02.png">

When y1 is equal to py, in the bounding box, any value of px greater than x1 returns true, hence in the bounding box it returns a straight line for the values of px greater than x value of node1.

I updated the if conditions to handle the condition y1 equal to py, rebuilt ITK and it is resolved for me.

Thanks,
Neslisah.
